### DISCUSSION TITLE
How do I implement Quill.js in my Uno Wasm?

### DISCUSSION THREAD

**Discussion by Ibrahim12404** [unoplatform/uno#16888] [Q&A] [answered]
I added <script src="https://cdn.jsdelivr.net/npm/quill@2.0.2/dist/quill.js"></script> before require.js in a custom index.html.

Then, I added the following class:

```csharp
[HtmlElement("div")]
public sealed partial class DivControl : FrameworkElement
{
    public DivControl()
    {
        Width = 500;
        Height = 500;
        this.SetHtmlAttribute("id", "editor");
    }
}
```

In wasm script, I added the js function (that's what Load Quill does in the video):

```javascript
function initializeQuill() {
    const quill = new Quill('#editor', {
        theme: 'snow'
    });
}
```

The quill library loads just fine. But, it appears corrupt and doesn't work. Somebody help me on how to add quill.js on my wasm.

https://github.com/unoplatform/uno/assets/97126964/13815063-7a10-437d-b8e9-e2a2ed6ec17f

**Comment by Ibrahim12404** [comment, marked as answer]
Never mind.

I forgot to add the css. But, it still had some issues.

I was then able to make it work just fine by using this.SetHtmlContent(), making a div and passing that div to quill. Is that a good idea? It works though.

Cool to see I can use all sorts of javascript libraries in my wasm app.

https://github.com/unoplatform/uno/assets/97126964/10476a9d-00d2-4425-9a8b-d309ae744c56

**Reply by jeromelaban** [reply]
Thanks for the update and thanks for using Uno Platform!
